app/models/product_importer.rb

```diff
@@ -177,7 +181,12 @@ def count_existing_products
           AND spree_variants.deleted_at IS NULL', supplier_id).
           count
 
-        @supplier_products[:by_supplier][supplier_id] = {existing_products: products_count}
+        if @supplier_products[:by_supplier][supplier_id]
+          @supplier_products[:by_supplier][supplier_id][:existing_products] = products_count
+        else
+          @supplier_products[:by_supplier][supplier_id] = {existing_products: products_count}
+        end
```

Comment:
So before this part (during validation) under @supplier_products[:by_supplier][supplier_id] there may or may not be a key set for [:updates_count] that counts the number of updated item per supplier. So here we need to check if the [supplier_id] key has been set already, because if it has then the [:existing_products] key is being set alongside the [:updates_count] key in the same level, and we don't want to overwrite this value.